authentication/http.go

```diff
@@ -93,31 +81,20 @@ func GetGroups(ctx context.Context) ([]string, bool) {
 	return groups, ok
 }
 
-// WithTenantMiddlewares creates a single Middleware for all
-// provided tenant-middleware sets.
-func WithTenantMiddlewares(middlewareSets ...map[string]Middleware) Middleware {
-	middlewares := map[string]Middleware{}
-
-	for _, ms := range middlewareSets {
-		for t, m := range ms {
-			middlewares[t] = m
-		}
-	}
-
+// Creates a single Middleware.
+func WithTenantMiddleware(middleware Middleware) Middleware {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			tenant, ok := GetTenant(r.Context())
+			_, ok := GetTenant(r.Context())
 			if !ok {
 				http.Error(w, "error finding tenant", http.StatusBadRequest)
 				return
 			}
-			m, ok := middlewares[tenant]
-			if !ok {
+			if middleware == nil {
```

Comment:
The check and the error message do not really align here. I don't think we really need to check that we were given a non-nil function pointer

Reply:
this code belongs to dead code / a function never used. removed the underlying  function.